src/LinkBubbleMenu/index.tsx

```diff
@@ -105,17 +105,40 @@ export default function LinkBubbleMenu({
             // full link "mark"
             .extendMarkRange("link")
             // Update the link href and its text content
-            .insertContent({
-              type: "text",
-              marks: [
-                {
-                  type: "link",
-                  attrs: {
-                    href: link,
-                  },
-                },
-              ],
-              text: text,
+            .command(({ tr, state }) => {
+              const existingHref = editor.isActive("link")
+                ? (editor.getAttributes("link").href as string)
+                : "";
+
+              const { selection, schema } = state;
+
+              if (existingHref) {
+                // Get the resolved position from the selection
+                const resolvedPos = state.doc.resolve(selection.from);
+                const nodeAfter = resolvedPos.nodeAfter;
+
+                if (nodeAfter?.isText) {
```

Comment:
Why is it we only use this logic below if specifically the next node is text? 

And what if there are multiple nodes within the selection, not just two?